### PR TITLE
bots: Add command to test PRs for npm dependencies

### PR DESCRIPTION
**Pull request by petervo**
Allows us to test PRs for npm dependencies before they get merged. Requires that the bot user has permissions on the remote npm repo as the github status results are written there.

### CODE REVIEW

**Comment by petervo** [comment]
While this can be reviewed, I need to get @cockpituous the needed credentials on openshift/registry-image-widgets before this can be merged.

**Comment by petervo** [comment]
Updated to use the repo in cockpit-project.

File: bots/npm-repo-test-invoke [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            "description": "{0} [{1}]".format(github.TESTING, HOSTNAME),
+            "target_url": ":link"
+        })
+
+
+    del os.environ["TEST_REVISION"]
+
+    os.environ["TEST_REVISION"] = head
+    os.environ["GITHUB_REVISION"] = revision
+    os.environ["GITHUB_KNOWN_ISSUE_BASE"] = git_base
```
Comment (petervo) [inline bots/npm-repo-test-invoke]: We have that. The problem is that when we set that variable to something that is not the main cockpit project, we still need to report known issues against the original repo.

**Comment by stefwalter** [comment]
I'm merging this. There will likely be follow ups.

**Pull request closed without merging**